googlemock/test/gmock-matchers_test.cc

```diff
@@ -2708,19 +2708,21 @@ class FloatingPointTest : public testing::Test {
         zero_bits_(Floating(0).bits()),
         one_bits_(Floating(1).bits()),
         infinity_bits_(Floating(Floating::Infinity()).bits()),
-        close_to_positive_zero_(
-            Floating::ReinterpretBits(zero_bits_ + max_ulps_/2)),
-        close_to_negative_zero_(
-            -Floating::ReinterpretBits(zero_bits_ + max_ulps_ - max_ulps_/2)),
+        close_to_positive_zero_(Floating::ReinterpretBits(
+            static_cast<Bits>(zero_bits_ + max_ulps_/2))),
+        close_to_negative_zero_(-Floating::ReinterpretBits(
+            static_cast<Bits>(zero_bits_ + max_ulps_ - max_ulps_/2))),
         further_from_negative_zero_(-Floating::ReinterpretBits(
-            zero_bits_ + max_ulps_ + 1 - max_ulps_/2)),
-        close_to_one_(Floating::ReinterpretBits(one_bits_ + max_ulps_)),
-        further_from_one_(Floating::ReinterpretBits(one_bits_ + max_ulps_ + 1)),
+            static_cast<Bits>(zero_bits_ + max_ulps_ + 1 - max_ulps_/2))),
+        close_to_one_(Floating::ReinterpretBits(
+            static_cast<Bits>(one_bits_ + max_ulps_))),
+        further_from_one_(Floating::ReinterpretBits(
+            static_cast<Bits>(one_bits_ + max_ulps_ + 1))),
         infinity_(Floating::Infinity()),
-        close_to_infinity_(
-            Floating::ReinterpretBits(infinity_bits_ - max_ulps_)),
-        further_from_infinity_(
-            Floating::ReinterpretBits(infinity_bits_ - max_ulps_ - 1)),
+        close_to_infinity_(Floating::ReinterpretBits(
+            static_cast<Bits>(infinity_bits_ - max_ulps_))),
+        further_from_infinity_(Floating::ReinterpretBits(
+            static_cast<Bits>(infinity_bits_ - max_ulps_ - 1))),
```

Comment:
Yeah, but the function’s argument needs to be templated because it’s not possible to know what it’s going to be. FloatingPointTest is instantiated with RawType = float and RawType = double, so Bits will be an unsigned 32-bit type or an unsigned 64-bit type. Meanwhile, size_t will be either 32 bits or 64 bits depending on the system. I don’t know how to express decltype(who_wins_the_promotion_race) without C++11.

I updated the pull request.